connector/src/main/java/org/geysermc/connector/GeyserConnector.java

```diff
@@ -190,6 +202,36 @@ private GeyserConnector(PlatformType platformType, GeyserBootstrap bootstrap) {
 
         defaultAuthType = AuthType.getByName(config.getRemote().getAuthType());
 
+        TimeSyncer timeSyncer = null;
+        if (defaultAuthType == AuthType.FLOODGATE) {
+            timeSyncer = new TimeSyncer(Constants.NTP_SERVER);
+            try {
+                Key key = new AesKeyProducer().produceFrom(config.getFloodgateKeyPath());
+                cipher = new AesCipher(new Base64Topping());
+                cipher.init(key);
+                logger.info(LanguageUtils.getLocaleStringLog("geyser.auth.floodgate.loaded_key"));
+                skinUploader = new FloodgateSkinUploader(this).start();
+            } catch (Exception exception) {
+                logger.severe(LanguageUtils.getLocaleStringLog("geyser.auth.floodgate.bad_key"), exception);
+            }
+        }
+        this.timeSyncer = timeSyncer;
+
+        String branch = "unknown";
+        int buildNumber = -1;
+        try {
+            Properties gitProperties = new Properties();
+            gitProperties.load(FileUtils.getResource("git.properties"));
+            branch = gitProperties.getProperty("git.branch");
+            String build = gitProperties.getProperty("git.build.number");
+            if (build != null) {
+                buildNumber = Integer.parseInt(build);
+            }
+        } catch (Exception e) {
```

Comment:
This doesn't seem to catch the `AssertionError` from `FileUtils.getResource()`

Reply:
It should catch that error now (in the 1.17 branch)